src/System.Memory/src/System/SpanHelpers.byte.cs

```diff
@@ -178,5 +178,76 @@ public static unsafe bool SequenceEqual(ref byte first, ref byte second, int len
         NotEqual: // Workaround for https://github.com/dotnet/coreclr/issues/9692
             return false;
         }
+
+        public static unsafe bool StartsWith(ref byte searchSpace, int searchSpaceLength, ref byte value, int valueLength)
+        {
```

Comment:
This maybe ok to duplicate - if you want to tune it differently from SequenceEqual. However, it should be still optimized to compare more than byte at a time.